### PR TITLE
Deliver VPC flow logs to a central account

### PR DESCRIPTION
**Pull request by zachmcclurg-ccpo**
open PR for kyle to critique my perfect code.

### CODE REVIEW

File: lib/atat-net-stack.ts [hunk 1]
```diff
@@ -55,10 +57,14 @@ export class AtatNetStack extends cdk.Stack {
...
 
+    const vflbucket = s3.Bucket.fromBucketAttributes(this, "ImportedBucket", {
+      bucketArn: props.vpcFlowLogBucket,
+    });
+
     // Capture all VPC flow logs and send to CloudWatch Logs with indefinite retention.
     // Flow log format made to meet C5ISR log format requirement
     /* eslint-disable no-template-curly-in-string */
-    vpc.addFlowLog("AllFlowLogs", {
+    vpc.addFlowLog("NewAllFlowLogs", {
```
[the remark this answers is not in the document]
Comment (zachmcclurg-ccpo) [reply lib/atat-net-stack.ts]: fixed

File: lib/atat-net-stack.ts [hunk 2]
```diff
@@ -82,13 +88,11 @@ export class AtatNetStack extends cdk.Stack {
...
-      destination: ec2.FlowLogDestination.toCloudWatchLogs(
-        new logs.LogGroup(this, "ALLFlowLogs", {
-          retention: logs.RetentionDays.INFINITE,
-        })
-      ),
+      destination: ec2.FlowLogDestination.toS3(vflbucket),
     });
 
+    // destination: ec2.FlowLogDestination.(
+
```
[the remark this answers is not in the document]
Comment (zachmcclurg-ccpo) [reply lib/atat-net-stack.ts]: fixed

[the remark this answers is not in the document]
Comment (zachmcclurg-ccpo) [reply lib/atat-net-stack.ts]: fixed

File: lib/atat-web-api-stack.ts [hunk 3]
```diff
@@ -31,6 +31,7 @@ export interface AtatWebApiStackProps extends cdk.StackProps {
   network?: AtatNetStack;
   isSandbox?: boolean;
   apiDomain?: ApiCertificateOptions;
+  vpcFlowLogBucket?: AtatNetStack;
```
[the remark this answers is not in the document]
Comment (zachmcclurg-ccpo) [reply lib/atat-web-api-stack.ts:34]: fixed

File: lib/atat-pipeline-stack.ts [hunk 4]
```diff
@@ -15,6 +15,7 @@ export interface AtatProps {
   vpcCidr?: string;
   notificationEmail?: string;
   apiDomain?: ApiCertificateOptions;
+  vpcFlowLogBucket?: AtatNetStack;
```
[the remark this answers is not in the document]
Comment (zachmcclurg-ccpo) [reply lib/atat-pipeline-stack.ts]: fixed

File: lib/atat-net-stack.ts [hunk 5]
```diff
@@ -1,11 +1,13 @@
 import * as cdk from "aws-cdk-lib";
 import * as ec2 from "aws-cdk-lib/aws-ec2";
+import * as s3 from "aws-cdk-lib/aws-s3";
 import * as logs from "aws-cdk-lib/aws-logs";
 import * as custom from "aws-cdk-lib/custom-resources";
 import * as route53Resolver from "aws-cdk-lib/aws-route53resolver";
 import { VpcDefaultSecurityGroupRuleRemover } from "./constructs/vpc-default-sg-rule-remove";
 
 import { Construct, DependencyGroup } from "constructs";
+import { Bucket } from "aws-cdk-lib/aws-s3";
```
[the remark this answers is not in the document]
Comment (zachmcclurg-ccpo) [reply lib/atat-net-stack.ts]: fixed

File: lib/atat-web-api-stack.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (juliusfitzhugh-ccpo) [reply lib/atat-web-api-stack.ts:34]: I think this one got missed by accident.

File: lib/atat-net-stack.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (juliusfitzhugh-ccpo) [reply lib/atat-net-stack.ts]: Should there be two calls to `vpc.addFlowLog`?

**Review by jasonburkert-ccpo — APPROVED** [review]
All of this looks reasonable.

**Pull request merged**